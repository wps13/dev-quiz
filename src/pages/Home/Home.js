import React from 'react';
import {SafeAreaView, View, Text, StyleSheet} from 'react-native';
import ButtonGeneric from '../../components/Button/Button';
import {baseStyles} from '../../assets/styles';

import COLORS from '../../utils/COLORS';

const Home = ({navigation}) => {
  return (
    <SafeAreaView style={styles.container}>
      <Text style={[styles.text, styles.title]}>DevQuiz</Text>
      <Text style={[styles.text, styles.subText]}>
        Test your knowledge about development
      </Text>
      <View style={styles.buttonWrapper}>
        <ButtonGeneric
          buttonColor={COLORS.darkPurple}
          text="Start"
          pressFunction={() => navigation.navigate('Question')}
          textColor={COLORS.white}
          testID="Home-ButtonQuestion"
        />
        <ButtonGeneric
          buttonColor={COLORS.lightPurple}
          text="Instructions"
          pressFunction={() => navigation.navigate('Instructions')}
          textColor={COLORS.darkGrey}
          testID="Home-ButtonInstructions"
        />
      </View>
    </SafeAreaView>
  );
};

export default Home;

const styles = StyleSheet.create({
  container: baseStyles.container,
  text: {
    margin: 10,
  },
  title: {
    fontSize: 28,
    fontWeight: '700',
  },
  subText: {
    width: '50%',
    textAlign: 'center',
  },
  buttonWrapper: {
    width: '100%',
    margin: 40,
    alignItems: 'center',
  },
});
