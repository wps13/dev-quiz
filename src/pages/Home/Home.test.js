import React from 'react';
import {shallow} from 'enzyme';

import Home from './Home';

const props = {
  navigation: {
    navigate: jest.fn(route => route),
  },
};

const renderHome = () => shallow(<Home {...props} />);

const questionsNode = wrapper =>
  wrapper.findWhere(node => node.prop('testID') === 'Home-ButtonQuestion');

const instructionsNode = wrapper =>
  wrapper.findWhere(node => node.prop('testID') === 'Home-ButtonInstructions');

describe('Home', () => {
  it('should render correctly', () => {
    const wrapper = renderHome();
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
  it('should render button to start the quiz', () => {
    const wrapper = renderHome();
    const questionsButton = questionsNode(wrapper);

    expect(questionsButton.exists()).toEqual(true);
    wrapper.unmount();
  });
  it('should render button to instructions', () => {
    const wrapper = renderHome();
    const instructionsButton = instructionsNode(wrapper);

    expect(instructionsButton.exists()).toEqual(true);
  });
  it('should navigate to questions on press Instructions button', () => {
    //montar componente
    const wrapper = renderHome();
    // simular press
    const questionsButton = questionsNode(wrapper);
    const route = questionsButton
      .first()
      .props()
      .pressFunction();
    // checar navigation
    expect(route).toEqual('Question');
    wrapper.unmount();
  });
  it('should navigate to  instructions  on press Start button', () => {
    //montar componente
    const wrapper = renderHome();
    // simular press
    const instructionsButton = instructionsNode(wrapper);
    const route = instructionsButton
      .first()
      .props()
      .pressFunction();
    // checar navigation
    expect(route).toEqual('Instructions');
    wrapper.unmount();
  });
});
