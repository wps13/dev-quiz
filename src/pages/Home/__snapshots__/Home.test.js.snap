// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "margin": 10,
        },
        Object {
          "fontSize": 28,
          "fontWeight": "700",
        },
      ]
    }
  >
    DevQuiz
  </Text>
  <Text
    style={
      Array [
        Object {
          "margin": 10,
        },
        Object {
          "textAlign": "center",
          "width": "50%",
        },
      ]
    }
  >
    Test your knowledge about development
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 40,
        "width": "100%",
      }
    }
  >
    <Button
      buttonColor="#A093F1"
      pressFunction={[Function]}
      testID="Home-ButtonQuestion"
      text="Start"
      textColor="#fff"
    />
    <Button
      buttonColor="#E7E4F9"
      pressFunction={[Function]}
      testID="Home-ButtonInstructions"
      text="Instructions"
      textColor="#302929"
    />
  </View>
</ForwardRef(SafeAreaView)>
`;
